src/main/java/org/jenkinsci/test/acceptance/recorder/TestRecorderRule.java

```diff
@@ -0,0 +1,147 @@
+import java.awt.Dimension;
+import java.awt.AWTException;
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * JUnit Rule that before executing a test it starts a recording current screen
+ * and after the test is executed, it stops recording.
+ */
+public class TestRecorderRule extends TestWatcher {
+
+    private static final Logger logger = LoggerFactory.getLogger(TestRecorderRule.class);
+
+    private static final int MAX_RECORDING_TIME_SECS = 120000;
+    private static final int FRAME_RATE_PER_SEC = 60;
+    private static final int BIT_DEPTH = 16;
+    private static final float QUALITY_RATIO = 0.97f;
+    public static final String TARGET = "target";
+
+    private JUnitScreenRecorder screenRecorder;
+
+    @Override
+    protected void starting(Description description) {
+        if (!isRecorderDisabled()) {
+            startRecording(description);
+        }
+    }
+
+    private void startRecording(Description des) {
+        GraphicsConfiguration gc = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice()
+                .getDefaultConfiguration();
+
+        File movieFolder = new File(TARGET);
+
+        String mimeType = FormatKeys.MIME_QUICKTIME;
+        String videoFormatName = VideoFormatKeys.ENCODING_QUICKTIME_ANIMATION;
+        String compressorName = VideoFormatKeys.COMPRESSOR_NAME_QUICKTIME_ANIMATION;
+        Dimension outputDimension = gc.getBounds().getSize();
+        int bitDepth = BIT_DEPTH;
+        float quality = QUALITY_RATIO;
+        int screenRate = FRAME_RATE_PER_SEC;
+        long maxRecordingTime = MAX_RECORDING_TIME_SECS;
+
+        Format outputFormatForScreenCapture = getOutputFormatForScreenCapture(videoFormatName,
+                compressorName, outputDimension,
+                bitDepth, quality, screenRate);
+
+        try {
+            this.screenRecorder = new JUnitScreenRecorder
+                    (gc, gc.getBounds(), getFileFormat(mimeType),
+                            outputFormatForScreenCapture, null, null, movieFolder, des);
+            this.screenRecorder.start();
+        } catch (IOException e) {
+            logger.warn("Exception starting test recording {}", e);
+        } catch (AWTException e) {
+            logger.warn("Exception starting test recording {}", e);
+        }
+    }
+
+    @Override
+    protected void succeeded(Description description) {
+        if (this.screenRecorder != null) {
+            if (saveAllExecutions()) {
+                stopRecordingWithFinalWaiting();
+            } else {
+                stopRecording();
+                this.screenRecorder.removeMovieFile();
+            }
+        }
+    }
+
+    @Override
+    protected void finished(Description description) {
+        stopRecordingWithFinalWaiting();
+    }
+
+    private boolean isRecorderDisabled() {
+        return Boolean.getBoolean("RECORDER_DISABLED");
```

Comment:
I have no real preference about this, but maybe it is true that positive is better, I am going to change it but then I need to do some tricky things since by default is enabled and I will need to check for null and so on in case of property is not set.